torch/csrc/distributed/c10d/ProcessGroup.cpp

```diff
@@ -170,6 +170,17 @@ void ProcessGroup::setGroupName(const std::string& name) {
   }
 }
 
+const std::string& ProcessGroup::getGroupDesc() const {
+  TORCH_CHECK(!deviceTypeToBackend_.empty(), "ProcessGroup desc not set");
+  return deviceTypeToBackend_.begin()->second->getGroupDesc();
```

Comment:
Hmm, are we going to the first backend to retrieve a ProcessGroup name?
Can't we make `pg_desc_` a member variable of `ProcessGroup`?

Reply:
here pg_desc_ follows the same logic as pg_name_. So I think make them consistent would be better?

Reply:
If `pg_name_` is done wrong, my recommendation is: do not follow it.
Later, someone can easily refactor `pg_name_` by following your right implementation.